README.md

```diff
@@ -4,49 +4,250 @@
+already set.
+
+```ts
+@customElement('my-element')
+class MyElement extends Polymer.Element {
+  ...
+}
+```
+
+### `@property`
+Define a Polymer property.
+
+`options` is a [Polymer property
+options](https://www.polymer-project.org/2.0/docs/devguide/properties) object.
+All standard options are supported, except for `value`; use a property
+initializer instead. If the [Metadata Reflection API](#metadata-reflection-api)
+is configured, the `type` option will be inferred from the TypeScript type and
+can be omitted.
+
+```ts
+@property({type: String, notify: true})
+foo: string = 'hello';
+```
+
+### `@computed`
+
+Define a [computed
+property](https://www.polymer-project.org/2.0/docs/devguide/observers#computed-properties).
+
+This decorator must be applied to a
+[getter](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/get),
+and it must not have an associated setter.
+
+```ts
+@computed('foo', 'bar')
+get fooBar() {
+  return this.foo + this.bar;
+}
+```
+
+For optional additional type saftey, pass your custom element class as a
+generic parameter. This allows TypeScript to check that all of your
+dependencies are valid properties.
+
+```ts
+@computed<MyElement>('foo', 'bar')
+get fooBar() {
+  return this.foo + this.bar;
+}
+```
+
+To define a computed property with more complex dependency expressions for
+which you may want to receive change values as arguments (e.g. sub-properties,
+splices, wildcards, etc.), or to set additional property options, define a
+standard property and set its `computed` option.
+
+```ts
+@property({computed: 'computeBaz(foo.*)', reflectToAttribute: true})
+baz: string;
+
+private computeBaz(fooChangeRecord: object) {
+  ...
+}
+```
+
+### `@observe`
+
+Define a [complex property
+observer](https://www.polymer-project.org/2.0/docs/devguide/observers#complex-observers).
+
+`targets` can be a single dependency expression, or an array of them. All
+observer dependency syntaxes are supported (property names, sub-properties,
+splices, wildcards, etc.).
+
+```ts
+@observe('foo')
+private fooChanged(newFoo: string) {
+  console.log(`foo is now: ${newFoo}`);
+}
+
+@observe(['foo', 'bar'])
+private fooBarChanged(newFoo: string, newBar: string) {
+  console.log(`foo is now: ${newFoo}, bar is now: ${newBar}`);
+}
+
+@observe('baz.*')
+private bazChanged(changeRecord: object) {
+  console.log('baz changed deeply');
+}
+```
+
+To define a [simple property
+observer](https://www.polymer-project.org/2.0/docs/devguide/observers#simple-observers),
+which receives both the old and new values, set the `observer` option on the
+property you want to observe to the observer name or (preferably) function
+reference.
+
+```ts
+@property({observer: MyElement.prototype.bazChanged})
+baz: string;
+
+private bazChanged(oldValue: string, newValue: string) {
+  console.log(`baz was: ${oldValue}, and is now: ${newValue}`);
+}
+```
+
+### `@query`
+
+Define a getter that calls
```

Comment:
Maybe say something along the lines of "replaces the field with a getter that..."

Reply:
Also, add some motivation: These allow you to easily declare the expected types of queries for type-checked usage without casting at the call sites.